src/display.cpp

```diff
@@ -2048,6 +2038,28 @@ bool display::set_zoom(unsigned int amount, const bool validate_value_and_set_in
 	return true;
 }
 
+bool display::validate_zoom_level_and_set_index(unsigned int& new_zoom, int &new_zoom_index)
```

Comment:
Please document the API contract of the new function: what the parameters should be set to by the caller, what does the function set them to when it returns, and what the return value means.

I know many other functions don't do this, but let's not accrue any more technical debt than we already have.

This function uses none of the members of `display`, so it doesn't need to be a member function of that class. It can be a local function in display.cpp instead.

Reply:
I try to name things in a self-explanatory way since blocks of comments aren't usually helpful and can become out-of-date (I class that as technical debt too). And since it's not used outside of the `display` class, that's why I made it private, but I suppose if it's Wesnoth convention to leave such things as local functions instead I'll do that. And also see about making some comments to your satisfaction.